include/wx/memoryallocator.h

```diff
@@ -0,0 +1,94 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        wx/memoryallocator.h
+// Purpose:     wxMemoryAllocator, wxMallocAllocator, wxNewAllocator
+// Author:      Aron Helser
+// RCS-ID:      $Id: $
+// Copyright:   (c) wxWidgets team
+// Licence:     wxWindows licence
+/////////////////////////////////////////////////////////////////////////////
+
+#ifndef _WX_MEMORYALLOCATOR_H_
+#define _WX_MEMORYALLOCATOR_H_
+
+#include "wx/defs.h"
+
+
+//-----------------------------------------------------------------------------
+// wxMemoryAllocator
+//-----------------------------------------------------------------------------
+
+class WXDLLEXPORT wxMemoryAllocator
+{
+public:
+    virtual ~wxMemoryAllocator() { }
+    
+    // may either return NULL or throw std::bad_alloc on error
+    virtual void *Alloc(size_t size) = 0;
+   
+    // should not throw, may be called with NULL pointer
+    virtual void Free(void *ptr) = 0;
+};
+
+
+//-----------------------------------------------------------------------------
+// wxMallocAllocator
+//-----------------------------------------------------------------------------
+
+class WXDLLEXPORT wxMallocAllocator: public wxMemoryAllocator
+{
+public:
+    // Will return NULL on error
+    virtual void *Alloc(size_t size) {
+        return malloc(size);
+    }
+   
+    // should not throw, may be called with NULL pointer
+    virtual void Free(void *ptr) {
+        free(ptr);
+    }
+
+    // singleton accessor:
+    static wxMemoryAllocator *Get();
+};
+
+//-----------------------------------------------------------------------------
+// wxNewAllocator
+//-----------------------------------------------------------------------------
+
+class WXDLLEXPORT wxNewAllocator : public wxMemoryAllocator
+{
+public:
+    // will trigger std::set_new_handler() first, then throw std::bad_alloc on
```

Comment:
Not sure if the comment (regarding std::set_new_handler()) here matches the code? I believe the code here is self-documentary.

Similarly for `wxNewNothrowAllocator::Alloc()`.